`# RETRY` markers are attached to wrong tests

[This test run](https://logs.cockpit-project.org/logs/pull-15406-20210223-175147-b8c9779b-rhel-8-4-bots-1710/log.html) from https://github.com/cockpit-project/cockpit/pull/15406 is rather hard to interpret:

![image](https://user-images.githubusercontent.com/200109/108950396-76fd8500-7666-11eb-8499-b0fa46e0f933.png)

[raw log](https://logs.cockpit-project.org/logs/pull-15406-20210223-175147-b8c9779b-rhel-8-4-bots-1710/log)

`TestKdump.testBasic` failed the first time, got retried, and succeeded then. But it did *not* get a `# RETRY` marker:
```
Traceback (most recent call last):
  File "/work/bots/make-checkout-workdir/test/verify/check-kdump", line 183, in testBasic
    self.assertNotEqual(
AssertionError: '' == ''

# Result testBasic (__main__.TestKdump) failed
# 1 TEST FAILED [248s on 3-cockpit-9]
not ok 43 test/verify/check-kdump TestKdump.testBasic# ----------------------------------------------------------------------
# testBootOrder (__main__.TestMachinesSettings)
[...]
# Result testBasic (__main__.TestKdump) succeeded
# 1 TEST PASSED [210s on 3-cockpit-9]
ok 43 test/verify/check-kdump TestKdump.testBasic
# ----------------------------------------------------------------------
# testSnapshotCreate (__main__.TestMachinesSnapshots)
```

A dozen lines down, `testBootOrder` succeeded, but was marked as retry:
```
# Result testBootOrder (__main__.TestMachinesSettings) succeeded
# 1 TEST PASSED [32s on 3-cockpit-9] # RETRY 1 (be robust against unstable tests)

ok 74 test/verify/check-machines-settings TestMachinesSettings.testBootOrder [ND@1]
```

In the same way, `TestShutdownRestart.testBasic` failed twice, and the third retry succeeded. Again without a `# RETRY` marker. But `TestRunTest.testRetry` ran in parallel, and got a `# RETRY` marker even though it passed:
```
# Result testBasic (__main__.TestShutdownRestart) failed
# 1 TEST FAILED [147s on 3-cockpit-9]
not ok 279 test/verify/check-system-shutdown-restart TestShutdownRestart.testBasic# ----------------------------------------------------------------------
# testRetry (__main__.TestRunTest)
# Result testRetry (__main__.TestRunTest) succeeded
# 1 TEST PASSED [108s on 3-cockpit-9] # RETRY 1 (be robust against unstable tests)

ok 288 test/verify/check-testlib TestRunTest.testRetry
# ----------------------------------------------------------------------
```

Somehow our tests now garble up the retry reasons and printing. This is fairly recent, maybe some fallout from commit a0c8d08f1067ca0614465eaf8bcb5447288621ca or similar?

--- a/test/common/run-tests
+++ b/test/common/run-tests
@@ -66,10 +66,10 @@ def print_test(test, print_tap=True, retry_reason=""):
             except BlockingIOError as e:
                 line = line[e.characters_written:]
                 time.sleep(0.1)
-    flush_stdout()
 
     if not print_tap:
         print(retry_reason)
+        flush_stdout()
         return
 
     print()  # Tap needs to start on a separate line

